scripts/experiment_2/nbconverted/4_similarity_analysis_hausdorff.py

```diff
@@ -0,0 +1,153 @@
+
+d = {}
+float_params = ["learning_rate", "kappa", "epsilon_std"]
+str_params = ["analysis_name", "NN_architecture"]
+lst_params = ["num_batches"]
+with open(config_file) as f:
+    for line in f:
+        (name, val) = line.split()
+        if name in float_params:
+            d[name] = float(val)
+        elif name in str_params:
+            d[name] = str(val)
+        elif name in lst_params:
+            d[name] = ast.literal_eval(val)
+        else:
+            d[name] = int(val)
+
+
+# In[3]:
+
+
+# Parameters
+analysis_name = d["analysis_name"]
+NN_architecture = d["NN_architecture"]
+num_batches = d["num_batches"]
+
+
+# In[4]:
+
+
+# Load data
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../.."))
+
+batch_dir = os.path.join(
+    base_dir,
+    "data",
+    "batch_simulated",
+    analysis_name)
+
+
+# In[5]:
+
+
+get_ipython().run_cell_magic('time', '', '# Calculate dissimilarity using Hausdorff distance\n\noutput_list = []\n\nfor i in num_batches:\n    print(\'Calculating hausdorff distance between 1 batch vs {} batches..\'.format(i))\n    \n    # Get batch 1 data\n    batch_1_file = os.path.join(\n        batch_dir,\n        "Batch_1.txt.xz")\n\n    batch_1 = pd.read_table(\n        batch_1_file,\n        header=0,\n        index_col=0,\n        sep=\'\\t\')\n\n    # Simulated data with all samples in a single batch\n    original_data_df =  batch_1\n    \n    # Get data with additional batch effects added\n    batch_other_file = os.path.join(\n        batch_dir,\n        "Batch_"+str(i)+".txt.xz")\n\n    batch_other = pd.read_table(\n        batch_other_file,\n        header=0,\n        index_col=0,\n        sep=\'\\t\')\n    \n    # Simulated data with i batch effects\n    batch_data_df =  batch_other\n    \n    # Calculate hausdorff distance\n    dist = max(directed_hausdorff(original_data_df, batch_data_df)[0],\n               directed_hausdorff(batch_data_df, original_data_df)[0])\n    \n    output_list.append(dist)')
+
+
+# In[7]:
+
+
+# Permute simulated data
+simulated_data_file = os.path.join(
+    base_dir,
+    "data",
+    "simulated",
+    analysis_name,
+    "simulated_data.txt.xz")
+
+# Read in data
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0, 
+    index_col=0,
+    sep='\t')
+
+simulated_data.head(10)
+
+shuffled_simulated_arr = []
+num_samples = simulated_data.shape[0]
+
+for i in range(num_samples):
+    row = list(simulated_data.values[i])
+    shuffled_simulated_row = random.sample(row, len(row))
+    shuffled_simulated_arr.append(shuffled_simulated_row)
+
+shuffled_simulated_data = pd.DataFrame(shuffled_simulated_arr, 
+                                       index=simulated_data.index, 
+                                       columns=simulated_data.columns)
+shuffled_simulated_data.head()
+
+
+# In[11]:
+
+
+# Hausdorff distance using permuted dataset (Negative control)
+permuted_dist = max(directed_hausdorff(original_data_df, shuffled_simulated_data)[0],
```

Comment:
why max?

Reply:
This was mainly from the documentation and follows from the definition of finding the max distance, I believe